python/SECRETSMANAGER_MAX_SECRET_AGE.py

```diff
@@ -0,0 +1,77 @@
+#
+# This file made available under CC0 1.0 Universal (https://creativecommons.org/publicdomain/zero/1.0/legalcode)
+#
+# Ensure that Secret LastChangedDate is no older than days specified
+# Description: Checks that Secrets have been rotated since number of days
+#
+# Trigger Type: Periodic
+# Scope of Changes: N/A
+# Accepted Parameters: max_secret_age_days
+# Example Values: 1, 2, 3,... default 30
+#
+# requires boto3 >= 1.9.152, botocore >= 1.12.152
+# will fail without these library versions as pagination
+# was not supported with the below libraries included
+# in Lambda as of May 21st 2019
+# Python 3.7 Lambda includes boto3 "1.9.42"
+# Python 3.6 Lambda includes boto3 "1.7.74"
+# You must use a custom deployment package that includes
+# the required boto3 and botocore versions
+# https://aws.amazon.com/premiumsupport/knowledge-center/build-python-lambda-deployment-package/
+
+import json
+from datetime import datetime, timedelta, timezone
+import boto3
+
+def evaluate_compliance(rule_parameters, secret):
+    if 'max_secret_age_days' in rule_parameters:
+        max_secret_age = datetime.now(timezone.utc) - timedelta(days=int(rule_parameters['max_secret_age_days']))
+    else:
+        max_secret_age = datetime.now(timezone.utc) - timedelta(days=30)
+
+    if 'LastRotatedDate' in secret and datetime.replace(secret['LastRotatedDate'], tzinfo=timezone.utc) > max_secret_age:
+        return "COMPLIANT"
+    if datetime.replace(secret['LastChangedDate'], tzinfo=timezone.utc) > max_secret_age:
```

Comment:
LastChangedDate is for all changes, including tags, policy, etc. You will need to look at list_secret_version_ids() to see the CreatedDate

Reply:
used versions instead